framework/rest/UpdateAction.php

```diff
@@ -45,7 +52,7 @@ public function run($id)
 
         $model->scenario = $this->scenario;
         $model->load(Yii::$app->getRequest()->getBodyParams(), '');
-        if ($model->save() === false && !$model->hasErrors()) {
+        if ($model->save(true, $this->attributeNames) === false && !$model->hasErrors()) {
```

Comment:
redundant property, use `Model::$scenario` to change active attributes

Reply:
Thanks for your comment. I know there is scenario, but i think scenarios are a good solution, but not the only one. In models with a lot of properties it gets messy very quickly. I prefer the proposed way, which won't hurt anyone, but a good inline documentation is necessary, i agree on this.

Reply:
@nadar Problem not "how", main problem in "where": when you declare attributes in actions you must updates action properties at changes in model structure - it's bad practice (not **S**olid). Can you explain cases when it's necessary?
Scenarios like as 'new/insert'(insert new record), 'public/frontend'(show in front-end) and 'default' cover 95% cases.

Reply:
We have a base library which does not know the scenarios where they are used. The code is completely decoupled. So we have multiple rest actions in different projects/applications or modules. 

If everyone thinks its a bad idea i will close it, but i think it would be a legit solution, easy to use, easy to understand and available if needed. In general, developers are mature and responsible enough to decide, give them the tools, not rules.

Reply:
@nadar such solution is difficult to maintain/debug. You can add scenarios for actions - it's give same result without above problems. By default, AR update only dirty/changed attributes, you declare attribute names to omit some changed attributes?

Reply:
After some thinking this indeed sounds like a duplicated functionality, especially that there is a scenario used by default in that action.